src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -353,20 +370,22 @@ const StudentApplication = ({ userProfile, authentication }) => {
    */
   const handleHallRemove = (index) => {
     if (index !== null && index !== -1) {
-      let newPreferredHalls = preferredHalls; // make a separate copy of the array
-      // Remove the selected hall if the list has more than one element
-      newPreferredHalls.splice(index, 1);
-
-      if (preferredHalls.length > 0) {
-        // If any rank value is greater than the new maximum, then set it to that new max rank
-        let maxRank = newPreferredHalls.length;
-        newPreferredHalls.forEach((hallInfo, index) => {
-          if (hallInfo.HallRank > maxRank) {
-            newPreferredHalls[index].HallRank = maxRank;
-          }
-        });
-      }
-      setPreferredHalls(newPreferredHalls);
+      setPreferredHalls((prevPreferredHalls) => {
+        let newPreferredHalls = prevPreferredHalls.filter((_hall, j) => j !== index);
+        // let newPreferredHalls = prevPreferredHalls.filter((hallInfo) => hallInfo.HallName !== hallNameToRemove);
```

Comment:
```suggestion
```

Reply:
I was considering that feature, basically to identify which thing to remove by using the name instead of the index, since indices change when the things gets sorted, though the feature works either way because the index of each `HallListItem` does get updated correctly every time the hall list changes, the same as how the rank numbers and hall names get updated correctly when you change the value selected in one of the dropdown menus.